resources/splunk_install.rb

```diff
@@ -0,0 +1,214 @@
+# Encoding: utf-8
+#
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_install
+#
+# Resource for managing the installation of Splunk
+class SplunkInstall < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::PathHelpers
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :version, String, required: true
+  property :build, String, required: true
+  property :base_url, String, default: 'https://download.splunk.com/products'
+
+  default_action :install
+
+  def after_created
+    node.run_state['splunk_ingredient'] ||= { 'installations' => {} }
+
+    case name.downcase
+    when 'splunk' then package :splunk
+    when 'universal_forwarder' then package :universal_forwarder
+    else raise 'Package must be specified (:splunk or :universal_forwarder)'
+    end unless property_is_set? :package
+  end
+
+  ### Inherited Methods
+
+  # Must be overridden by a platform provider
+  def kernel_string
+    raise 'No implementation for current platform'
+  end
+
+  def package_name
+    @package_name ||= package_names[package][node['os'].to_sym]
+    raise "Unsupported Combination: #{package} + #{node['os']}" unless @package_name
```

Comment:
Follow up question, which one of these raise statements does that test hit?  I think this one is dead code since the install_dir method is currently always called before this one (if I'm manually tracing the code paths correctly).